src/autolabel/task_chain.py

```diff
@@ -0,0 +1,363 @@
+from collections import defaultdict
+import copy
+from itertools import groupby
+import uuid
+from autolabel.configs import AutolabelConfig
+import logging
+from typing import Dict, List, Optional
+
+from autolabel.labeler import LabelingAgent
+from autolabel.dataset import AutolabelDataset
+from autolabel.few_shot import (
+    BaseExampleSelector,
+)
+from autolabel.cache.sqlalchemy_generation_cache import SQLAlchemyGenerationCache
+from autolabel.cache.sqlalchemy_transform_cache import SQLAlchemyTransformCache
+from autolabel.cache.sqlalchemy_confidence_cache import SQLAlchemyConfidenceCache
+from autolabel.cache.base import BaseCache
+from pydantic import BaseModel
+from autolabel.configs import TaskChainConfig
+from autolabel.schema import TaskType
+from autolabel.transforms import TransformFactory
+from transformers import AutoTokenizer
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+logging.getLogger("httpx").setLevel(logging.WARNING)
+
+
+class ChainTask(BaseModel):
+    type: str
+    id: str
+    name: str
+    input_columns: List[str]
+    output_columns: List[str]
+    autolabel_config: AutolabelConfig = None
+    config: Dict = None
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class TaskGraph:
```

Comment:
https://docs.python.org/3/library/graphlib.html -> might be useful for doing topo-sort?